### ISSUE TITLE
Allow optional Step Title when adding a step

### ISSUE THREAD

**Issue by palerdot** [microsoft/codetour#181]
The default code path as default title is not that helpful. It would be great if we there is a way to add a title when adding a tour step.

**Comment by lostintangent** [comment]
When you add a step, you can include a markdown heading in the content and that will represent the title ([details](https://github.com/microsoft/codetour#step-titles). That's what many users use, since it doesn't require you to define a separate title, aside from the content itself. This is also nice, since you can just click a line of code and add the title/content to it in one step, as opposed to needing to define a title and then the content.

Would this work for your use cases?

**Comment by palerdot** [comment]
> When you add a step, you can include a markdown heading in the content and that will represent the title

This is nice, I overlooked this part in the docs.

I tried this approach. But there is a **problem, where title is not in sync with the first line of the content.** When you right click on the title and edit the title, the content is not actually updated. For example, I frequently edit the tour title based on my current understanding. But the updated title is not reflecting in the first line of the content. This kind of makes the content stale. 

It would be nice, if there is a two way sync, like `Step title <==> markdown heading in the content `. Right now, it is mostly useful to create a good initial step title.

**Comment by lostintangent** [comment]
Hmm, that two-way sync is the intention, and so if it's not working, that's just a bug :D Let me take a look at what's going on. If you reload VS Code, does that update the titles you see in the tree? I'm guessing it should, and this is just a matter of updating the title immediately after editing it.

**Comment by palerdot** [comment]
> If you reload VS Code, does that update the titles you see in the tree?

No. The changed title is not reflecting in the first line of the content even after reloading.

**Comment by lostintangent** [comment]
This should now be fixed. Thanks!
